### ISSUE TITLE
Cannot source SymLink pointing to file with different extension

### ISSUE THREAD

**Issue by david-cortes** [RcppCore/Rcpp#926]
In Linux, if I create a symlink of a file, and give that symlink a file name ending in `.cc` or `.cpp`, the function `sourceCpp` will still refuse to source it if the original file (to which the symlink points) had a different extension such as `.c`.

**Comment by eddelbuettel** [comment]
Ok. So?

I mean, seriously?  Can't you copy the file to a tempfile name `whateveryourwantplusrandom.cpp` and source that?

**Comment by eddelbuettel** [comment]
It's the old "if it is important to you" then maybe you want to trace this and see if you can add an extra flag or override.

`sourceCpp()` is a little special in that it is for interactive use.  You have yet to demonstrate why you input has to be a symlink.  Also note that you can get the _content_ in front of the compiler in other ways -- ie via already-parsed text.

At the end of the day, we care most about package support where R itself defines what is allowed in `src/` -- I would suspect (though I am claiming to know for sure) that symlinks are a no-no there.  Besides saying "it doesn't work" (to which I replied "then don't do it" per the old doctor joke) you have not really demonstrated why it is needed or what it would make better.  These file extensions are _used_ to determine the file type.  That is how `make` and many other tools work.  And a C file is not a C++ file.  Now many C _source_ files can be recast as C++ _source_ files but that requires that extension.

In short I am still inclined to close this as a non-issue.

**Comment by david-cortes** [comment]
Well then, I guess I’ll better stick to reading the file into characters.

As for why I wanted to use symlinks, I had a situation like this: I’m editing C files that are to be used from within different software. For testing them in R, the `sourceCpp` function provides a very convenient way of reloading them. Since they end in `.c`, I thought of the symlinks as a quicker way of sourcing them. But that’s just my use case and I guess reading into characters will do just fine.

**Comment by eddelbuettel** [comment]
How about creating a file `foo.cpp` containing one line `#include "foo.c"` (plus maybe the required `extern "C" ...` business if that really is a plain C file).   Allows you to have your file, and let's the compiler see a cpp file.
